### ISSUE TITLE
Cells with periods and commas seem to parse incorrectly

### ISSUE THREAD

**Issue by jeffbonasso** [Khan/simple-markdown#32]
If there is a simple table with decimals and commas, it seems to create new lines for them...

This simple table...
Markdown | Test
--- | ---
Periods | Commas
25.7 | 1,000

Produces...
![image](https://user-images.githubusercontent.com/5274224/30248251-0ef5e28e-95f2-11e7-8c56-c14af45833e5.png)

**Comment by ariabuckles** [comment]
Hi,

Are you using the default react or html outputters / `reactFor` or `htmlFor`, or something else?

I can't reproduce this on stock simple-markdown, but I have a theory about what could be going on. My guess is that what is happening here is that it is parsing that as two separate text elements, which is (suprisingly) what it should be doing. The `reactFor` outputter is where it joins those into one output. Even if you didn't have that, they should show up as spans next to each other. If you aren't using `defaultReactOutput` or `reactFor` to join your rules, then you might be getting multiple spans next to each other, which the browser might wrap like this? or perhaps those spans have an odd `display:` style?

If you have the code you're using to create your outputter, or a library you're using which you can point me towards, that would be helpful. Thanks!

**Comment by jeffbonasso** [comment]
Thanks for the insight! I am utilizing another library called
react-native-simple-markdown which has the dependency on simple-markdown. I
traced it and yes in nestedParse function when it gets to text type it
splits it in two and pushes on an array. My guess is the
react-native-simple-markdown is then not handling appropriately and the
problem probably lies there. Looking in that code it is calling...

return SimpleMarkdown.reactFor(SimpleMarkdown.ruleOutput(rules, 'react'
))(tree)

...so maybe something else going on. Maybe it is how React Native is
handling the spans?

**Comment by ariabuckles** [comment]
Thanks @jeffbonasso ; i've got a test site for that library locally and I'll try to see what could be going on.

**Comment by ariabuckles** [comment]
For anyone following along:

This looks like it's caused because each text node in react native is outputting a separate Text component for each text node, since they aren't just strings like they are in web react.

I need to think for a day or two about the best way to solve this, but this should be solvable.

In the meantime, this particular result might be caused by odd style rules for table cells or rows/columns? it's just two Text nodes text to each other, which normally would appear next to each other, but might be wrapping due to styles? (on my test code, they don't wrap separately like that). (I'll still work on addressing the root problem here, of course)

**Comment by f0rr0** [comment]
Any updates on this @ariabuckles @jeffbonasso ?

**Comment by ariabuckles** [comment]
Thanks for the ping @f0rr0 & sorry for the delay on this @jeffbonasso . I've got a lot of work that refactors significantly how simple-markdown works to let things like this be easier.

It's not quite ready yet, but work has been ridiculously busy lately, so I've published what I have (which is working, but not well documented) as 1.0.0-alpha.0, which you can install via:

```
npm install --save simple-markdown@next
```

This changes the API a bit, some of the changes of which are documented in README.md (working on it), but the main change you'd need to do to get alternative text node joining behaviour (which fixes this) is to use the new `outputFor(...)` instead of `reactFor(ruleOutput(...))`, as follows:

```
var output = outputFor(myRules, "react");
```

instead of

```
var output = reactFor(ruleOutput(myRules, "react"));
```

**Comment by ariabuckles** [comment]
For anyone interested, the above API is now released, in version 0.4.0. It's not well documented yet, but it should fix this problem.

Basically, to fix this:

`npm install simple-markdown@latest`

And then change

```javascript
var output = reactFor(ruleOutput(myRules, "react"));
```

to

```javascript
var output = outputFor(myRules, "react");
```
